core/shared/src/main/scala/fs2/internal/CompileScope.scala

```diff
@@ -516,27 +510,39 @@ private[internal] object CompileScope {
   private object InterruptContext {
 
     /**
-      * Creates a new interrupt context for a new scope if the scope is interruptible.
+      * Creates a new interrupt context for a new scope with the given Concurrent.
       *
       * This is UNSAFE method as we are creating promise and ref directly here.
       *
       * @param interruptible  Whether the scope is interruptible by providing effect, execution context and the
       *                       continuation in case of interruption.
       * @param newScopeId     The id of the new scope.
+      * @param cancel         the parent cancellation method for the new scope.
       */
-    def unsafeFromInteruptible[F[_]](
-        interruptible: Option[Concurrent[F]],
-        newScopeId: Token
-    )(implicit F: Sync[F]): Option[InterruptContext[F]] =
-      interruptible.map { concurrent =>
-        InterruptContext[F](
-          concurrent = concurrent,
-          deferred = Deferred.unsafe[F, Either[Throwable, Token]](concurrent),
-          ref = Ref.unsafe[F, Option[Either[Throwable, Token]]](None),
-          interruptRoot = newScopeId,
-          cancelParent = F.unit
-        )
-      }
+    def unsafeFromConcurrent[F[_]: Sync](
+        concurrent: Concurrent[F],
+        scopeId: Token,
+        cancel: F[Unit]
+    ): InterruptContext[F] =
+      InterruptContext[F](
+        concurrent = concurrent,
+        deferred = Deferred.unsafe[F, Either[Throwable, Token]](concurrent),
+        ref = Ref.unsafe[F, Option[Either[Throwable, Token]]](None),
+        interruptRoot = scopeId,
+        cancelParent = cancel
+      )
+
+  }
+
+  /* A Lease for the compile scope, which is just an aggregation of leases */
+  private class Lease[F[_]](leases: Chain[Scope.Lease[F]])(implicit F: Applicative[F])
```

Comment:
@mpilquist Per our recent conversation: the idea of a _Lease_ made of many other leases, is it general enough to merit  adding it to the `fs2.Scope` companion object?

Reply:
Not public at least, since I don't think we'd want to expose the underlying chain to users.